cli/js/io.ts

```diff
@@ -49,13 +49,95 @@ export interface SeekerSync {
   seekSync(offset: number, whence: SeekMode): number;
 }
 
+class ReadableStreamIOReader implements Reader {
+  #reader: ReadableStreamDefaultReader;
+  #buffer: Uint8Array | null;
+  #encoder: TextEncoder;
+  constructor(dst: ReadableStream) {
+    this.#reader = dst.getReader();
+    this.#buffer = null;
+    this.#encoder = new TextEncoder();
+  }
+
+  async read(p: Uint8Array): Promise<number | null> {
+    let value = this.#buffer;
+    if (value === null) {
+      // if buffer is empty, read from the stream
+      const res = await this.#reader.read();
+      if (res.done) {
+        this.#reader.releaseLock();
+        return null; // EOF
+      }
+
+      value = res.value;
+
+      if (!value) {
+        return 0;
+      }
+
+      if (typeof value === "string") {
+        value = this.#encoder.encode(value);
+      }
+    } else if (this.#buffer !== null && p.length >= this.#buffer.length) {
+      // If buffer size is lower or equal than view size, set buffer to null
+      this.#buffer = null;
+    }
+
+    if (value.length > p.length) {
+      // If chunk is bigger than view buffer exceeding bytes
+      this.#buffer = value.subarray(p.length);
+      value = value.subarray(0, p.length);
+
+      if (!this.#buffer.length) {
+        this.#buffer = null;
+      }
+    }
+
+    p.set(value, 0);
+    return value.length;
+  }
+}
+
+class WritableStreamIOWriter implements Writer, Closer {
+  #writer: WritableStreamDefaultWriter;
+  constructor(dst: WritableStream) {
+    this.#writer = dst.getWriter();
+  }
+
+  async write(p: Uint8Array): Promise<number> {
+    await this.#writer.ready;
+    await this.#writer.write(p);
+
+    return p.length;
+  }
+
+  close(): void {
+    return this.#writer.releaseLock();
+  }
+}
+
+export function toWriter(dst: WritableStream): Writer & Closer {
+  return new WritableStreamIOWriter(dst);
+}
+
+export function toReader(src: ReadableStream): Reader {
+  return new ReadableStreamIOReader(src);
+}
```

Comment:
I guess those utils are useful but exposing them as `Deno.toWriter()` and `Deno.toReader()` doesn't seem to be a good idea

Reply:
My initial version was only for internal use but thought users will need a way to convert between Streams & Reader/Writer. Don't mind making them internal-only again.